libsplinter/src/node_registry/mod.rs

```diff
@@ -137,12 +132,8 @@ where
         (**self).add_node(node)
     }
 
-    fn update_node(
-        &self,
-        identity: &str,
-        updates: HashMap<String, String>,
-    ) -> Result<(), NodeRegistryError> {
-        (**self).update_node(identity, updates)
+    fn update_node(&self, identity: &str, node: Node) -> Result<(), NodeRegistryError> {
```

Comment:
If you're passing the node to this API, do you still need to pass identity?  Then you wouldn't need the error for identity mismatch.

Reply:
Updated to add/update nodes in the registry using a single `add_or_update_node` method. The REST endpoints have been updated accordingly, with the update endpoint returning an error if the identity of the node is changed.